### ISSUE TITLE
Question: promote a deploy as the main deploy?

### ISSUE THREAD

**Issue by rstacruz** [nwtgck/actions-netlify#204]
Awesome work on actions-netlify! I'm really loving this and I'm very thankful that this exists.

Is it possible to make a deployment the main one that appears in `my-site.netlify.app`? When I try to set this up, the URL appears as `<alias>--my-site.netlify.app`.

**Comment by nwtgck** [comment]
@rstacruz Hi!

Your "main deployment" is called "production deployment" in this action. You can use `production-branch: master` to deploy every commit on master branch  to my-site.netlify.app. You can also specify `production-deploy: ${{ some condition }}` to control the "production deployment", so `production-deploy: true` means every commit to every branch deploys my-site.netlify.app.

To know the reason why "production deployment" is used, you can see the discussion on https://github.com/nwtgck/actions-netlify/issues/102#issuecomment-625462180.

**Comment by rstacruz** [comment]
Thanks! I tried setting production-branch: master and production-deploy: true, but it didn't quite work (https://github.com/rstacruz/til/blob/d152a20914842fc758ffbf67ab713bce7ed2e091/.github/workflows/build.yml). I'll try to investigate some more.

**Comment by nwtgck** [comment]
@rstacruz I can see https://rstacruz-tilnext.netlify.app on the commit comment. Is it the main deploy?

<img width="853" alt="image" src="https://user-images.githubusercontent.com/10933561/85934683-66530e80-b921-11ea-988a-208d60dbde3a.png">
(from: https://github.com/rstacruz/til/commit/d152a20914842fc758ffbf67ab713bce7ed2e091)

**Comment by rstacruz** [comment]
Yeah - however I think it's not showing up because I turned off builds in Netlify maybe? `rstacruz-tilnext.netlify.app` still shows an old version right now 🤔, though the other URL is okay.

I think this might be a problem on my end of configuration, don't worry too much, just thought I'd let you know the caveats I'm running into :)

**Comment by rstacruz** [comment]
I think this may be the source of my issue. Netlify has a "auto publishing" feature under the **Deploys** tab.

![image](https://user-images.githubusercontent.com/74385/85936085-82af7500-b93a-11ea-8ad9-6faa96f5a5f0.png)

![image](https://user-images.githubusercontent.com/74385/85936095-8b07b000-b93a-11ea-9ef4-9e7125c7c316.png)

**Comment by nwtgck** [comment]
@rstacruz Unfortunately, I could understand well your situation. Could you tell me "Production branch" in the "Deploy contexts" on "Build & deploy". Here is my setting.
<img width="781" alt="image" src="https://user-images.githubusercontent.com/10933561/85936323-85a76700-b934-11ea-8ca5-99fc349a2869.png">

Navigation: `Settings > Build & deploy > Deploy contexts`

**Comment by rstacruz** [comment]
Oh hilarious, I just did a very similar thing just now (set it to a non-existent branch)! I'll play around with it and report back when I get to my laptop.

**Comment by rstacruz** [comment]
I took some screenshots of my settings. Might help with updating documentation.

![image](https://user-images.githubusercontent.com/74385/85938887-805a1480-b954-11ea-8c73-421d86c11334.png)

![image](https://user-images.githubusercontent.com/74385/85938916-bac3b180-b954-11ea-90cf-a5ce43d0242c.png)

**Comment by rstacruz** [comment]
I was looking at the netlify cli and found that you can't specify alias and prod at the same time.

```
› yarn netlify deploy --dir public --alias v2020 --prod --message 'Manual'

   Error: --prod and --alias flags cannot be used at the same time
```

**Removing `alias` from the configuration works** - the deployment now becomes the "main" deploy :)

**Comment by rstacruz** [comment]
Would you accept a PR to throw an error when `alias` is used when `production-deploy: true` is also given?

**Comment by nwtgck** [comment]
@rstacruz First, thank you so much for your reporting and investigation! Deployment related to "production" deploy is very important thing.

> Would you accept a PR to throw an error when alias is used when production-deploy: true is also given?

Here are candidates to solve this problem when  `alias: ...` and `draft === false` (internal implementation, it means "production deploy").

* Throw error
* Ignore "alias" deploy and do "production" deploy only
* Do both "alias" deploy and "production" deploy (run `await netlifyClient.deploy()` twice internally)

"Throw error" is good to notify users and correct their usages. But, fail of CI might make GitHub Actions user uncomfortable because "production" deploy, which is very important and is triggered in a specific condition usually.

"Ignore alias" and notify users with warning is also a candidate not to fail the CI and deploy the important "production".

Deploying both can be our expectation, so although run `await netlifyClient.deploy()` twice internally, it may be what we should.

**Comment by nwtgck** [comment]
Should we change the title?

**Comment by rstacruz** [comment]
> Should we change the title?

Feel free to edit the title to what you feel is best :)

**Comment by rstacruz** [comment]
The behaviour is up to you, but my personal preference is to throw an error.

- it's least surprising for those who use netlify-cli since it mirrors its behaviour
- deploying twice will make 2 deploys show up, making it a bit surprising

My second preference would be to let it through and show a warning ("warning: ignoring 'alias'. Aliases are not supported in production mode").

... Just my 2 cents as a casual netlify user - I'd be happy with whatever you decide :)

**Comment by devotox** [comment]
@rstacruz @nwtgck - Just ran into this exact problem!

I would say if its a production build then it does not need to be built twice. just ignore the alias and not show the second URL either

**Comment by nwtgck** [comment]
@rstacruz @devotox 

Thank you for your patience. I'll solve this problem in #260 .

**Comment by devotox** [comment]
@nwtgck Wow you are quick!

**Comment by nwtgck** [comment]
@rstacruz @devotox 

I released 1.1.6 to fix this problem. Let me know when you find a problem!
